spec/stack.js

```diff
@@ -21,35 +21,49 @@ describe('Stack', () => {
   })
 
   context('pop()', () => {
-    it('returns and removes the top element in a stack or null if stack is empty', () => {
-      const myStack = new Stack()
+    context('when the stack is not empty', () => {
+      it('returns and removes the top element in a stack', () => {
+        const myStack = new Stack()
+
+        myStack.push('popcorn')
+        expect(() => myStack.pop())
+          .to.alter(() => myStack.length(), { from: 1, to: 0 })
+      })
+    })
+
+    context('when the stack is empty', () => {
+      it('returns null', () => {
+        const myStack = new Stack()
 
-      myStack.push('popcorn')
-      expect(() => myStack.pop())
-        .to.alter(() => myStack.length(), { from: 1, to: 0 })
+        expect(myStack.pop())
+          .to.equal(null)
+      })
     })
   })
 
   context('peek()', () => {
-  it('returns the top element in the stack or null if the stack is empty', () => {
+    it('returns the top element in the stack or null if the stack is empty', () => {
```

Comment:
This test is attempting to test two things; consider creating two tests that test each of the cases identified in the description:
1. returns top element of the stack
1. null if the stack is empty